TowerOfHanoi/Launcher.java

```diff
@@ -47,16 +51,22 @@ public static void main(String[] args){
             int toTargetArray;
             int valueToMove = 0;
 
-            // VYH: Validate your input over the first 3 chars, ignore anything else after that ~ CAL: done
-            // validate that the actual input was # # (char space char) ~ CAL: done
             try {
-                if (merp.length >= 3) {
-                    // VYH: Sanitize should return an int 0 1 or 2 if a valid character was entered ~ CAL: done
-                    // VYH: Make Sanitize throw an exception if input is not valid ~ CAL: done
-                    // VYH: catch exception here and move forward with next iteration ~ CAL: done
+                if (x.equalsIgnoreCase("undo")){
+                    Integer moveFrom = moveHistory.pop();
+                    Integer moveTo = moveHistory.pop();
+                    if (moveFrom != null && moveTo != null){
+                        System.out.println(moveMe(stacks[moveFrom], stacks[moveTo]));
+                        currentTurn++;
+                    } 
+                } else if (merp.length >= 3) {
```

Comment:
This is still not validating the input to be `# #` (char space char). I can still enter `aab` and will get a valid move

Reply:
This has been corrected.  After it makes sure the undo or exit command wasn't entered, it checks to see if the move entered is 3 or less char in length and that the second char is equal to the int value of "space".